pkg/cloud/common/templates.go

```diff
@@ -0,0 +1,82 @@
+package common
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	"sigs.k8s.io/yaml"
+	"text/template"
+
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type TemplateSource struct {
+	Object client.Object
+	Path   string
+}
+
+type ObjectTemplate struct {
+	TemplateSource
+	Template *template.Template
```

Comment:
I do think that template should not be in the structure, and just be an internal way to read embed fs.